### DISCUSSION TITLE
How to use both port and securePort in CephObjectStore when both are specified

### DISCUSSION THREAD

**Discussion by ushitora-anqou** [rook/rook#12613] [Q&A] [answered]
Currently, I provide non-secure RGW endpoints and am trying to support TLS endpoints too. It's better for the existing users to continue to use the non-secure endpoints without any modification.

To achieve the aforementioned goal, I specified both port and securePort (*1). However, I encountered the following problems.

a. How to use both non-secure ports and TLS ports is not documented.
b. If securePort is set, the `BUCKET_PORT` environment variables in ConfigMaps corresponding to the existing OBCs, are changed to use a TLS endpoint.

Problem b is caused by [this code.](https://github.com/rook/rook/blob/cb44c7b88fb346f3b120e8bda769ac808ec60880/pkg/apis/ceph.rook.io/v1/object.go#L49) I found that non-secure endpoints can be used by creating new StorageClasses having endpoint field. However, I hesitate to adopt this approach because of the following reasons.

- The endpoint field [seems to be for external clusters](https://github.com/rook/rook/commit/b86bb9c5286d7bc953848907be0dee7bccb3c27c).
- I don't want to write the endpoint's IP address directly in StorageClass.

(*1) According to [the ObjectStore design docs](https://github.com/rook/rook/blob/5979b3816292df6b3ff17eb8e58eb8cf88488802/design/ceph/object/store.md#gateway), both port and securePort can be set.

**Comment by parth-gr** [comment, marked as answer]
You can move to rook issue, if there is a bug

**Reply by ushitora-anqou** [reply]
At least, I think it is a bug that the documentation does not show how to use both port and securePort. I will move this to Issue. Thank you.

Issue URL: https://github.com/rook/rook/issues/12614
